src/utils/encoding.ts

```diff
@@ -0,0 +1,56 @@
+import fs from "fs";
+import iconv from "iconv-lite";
+import readline from "readline";
+import { extractFileFormat } from "./file";
+import type { SupportedImportEncoding } from "./file";
+
+export const isMismatchEncoding = async (
+  filePath: string,
+  encoding: SupportedImportEncoding,
+): Promise<boolean> => {
+  const format = extractFileFormat(filePath);
+  let isMismatch = false;
+  switch (format) {
+    case "csv":
+      isMismatch = await isMismatchEncodingOfCsvFile(filePath, encoding);
+      break;
+  }
+
+  return isMismatch;
```

Comment:
You can return in switch block
```suggestion
      return await isMismatchEncodingOfCsvFile(filePath, encoding);
  }

  return false;
```